server/athenian/api/align/mutations/goals.py

```diff
@@ -73,18 +108,76 @@ def _parse_create_goal_input(input: Dict[str, Any], account_id: int) -> GoalCrea
 def _parse_team_goal_input(team_goal_input: dict) -> TeamGoal:
     """Parse TeamGoalInput into a Team model."""
     try:
-        # get the first non null value in GoalTargetInput
-        target = next(tgt for tgt in team_goal_input["target"].values() if tgt is not None)
+        target = _parse_team_goal_target(team_goal_input["target"])
     except StopIteration:
         raise GoalMutationError(f'Invalid target for teamId {team_goal_input["teamId"]}')
 
     return TeamGoal(team_id=team_goal_input["teamId"], target=target)
 
 
+def _parse_team_goal_target(team_goal_target: dict) -> Union[int, float, str]:
+    """Get the first non null value in GoalTargetInput."""
+    return next(tgt for tgt in team_goal_target.values() if tgt is not None)
+
+
 def _convert_goal_dates(valid_from: date, expires_at: date) -> Tuple[datetime, datetime]:
     """Convert date objects from API into datetimes."""
     valid_from = datetime.combine(valid_from, time.min, tzinfo=timezone.utc)
     # expiresAt semantic is to include the given day, so datetime is set to the start of the
     # following day
     expires_at = datetime.combine(expires_at + timedelta(days=1), time.min, tzinfo=timezone.utc)
     return (valid_from, expires_at)
+
+
+@dataclass(frozen=True)
+class GoalUpdateInfo:
+    """The information to update an existing Goal."""
+
+    team_goal_deletions: Sequence[int]
+    team_goal_assignments: Sequence[TeamGoalTargetAssignment]
+
+
+def _parse_update_goal_input(input: Dict[str, Any], account_id: int) -> GoalUpdateInfo:
+    deletions = []
+    assignments = []
+
+    duplicates = []
+    both = []
+    invalid_targets = []
+    seen = set()
+    for change in input.get("teamGoalChanges", ()):
```

Comment:
Same here, we shouldn't hardcode "teamGoalChanges" string, it should exist inside a model.